### DISCUSSION TITLE
Create View in background thread

### DISCUSSION THREAD

**Discussion by georgemark0v** [NativeScript/NativeScript#9644] [Q&A]
It's possible to create views not in UI thread?
Im have chat messages list where im use recyclerview. When I scroll to the start of the story, the app freezes on every view creation. I want the views to be created on a separate thread and not slow down scroll. 

Video: https://yadi.sk/i/nIPJtfCw_tRvqA

**Comment by farfromrefug** [comment]
@georgemark0v you cant create view in a background thread. However what you are seeing is a real issue with N. View creation is far too slow. I have been investigating that for quite some time. 
It is important to know it depends a lot on the vieelws you are using in your cells. But for example labels creation should not be so slow. 
One reason I think is CSS. 
To make things faster for now:
* use as less many views as possible
* don't use spans 
* use as less CSS as possible.

If you need a lot of span and labels I suggest using @nativescript-community/canvaslabel. I always do that now.
Also you can even directly "draw" your cells content with @nativescript-community/canvas
This is what telegram do (the native way) in their app

**Comment by triniwiz** [comment]
A couple things

**Comment by georgemark0v** [comment]
> @georgemark0v you cant create view in a background thread. However what you are seeing is a real issue with N. View creation is far too slow. I have been investigating that for quite some time.
> It is important to know it depends a lot on the vieelws you are using in your cells. But for example labels creation should not be so slow.
> One reason I think is CSS.
> To make things faster for now:
> 
> * use as less many views as possible
> * don't use spans
> * use as less CSS as possible.
> 
> If you need a lot of span and labels I suggest using @nativescript-community/canvaslabel. I always do that now.
> Also you can even directly "draw" your cells content with @nativescript-community/canvas
> This is what telegram do (the native way) in their app

Thanks for your reply. Canvas/CanvasLabel supports self sizing?

**Comment by farfromrefug** [comment]
@georgemark0v not yet know. You have dynamic size cells?

**Comment by georgemark0v** [comment]
> @georgemark0v not yet know. You have dynamic size cells?

Not in the chat list. But messages of course have cells of different sizes depending on the size of the text. 
this low performance im only see on android.

**Comment by farfromrefug** [comment]
Yes the issue is mostly on android.

**Comment by triniwiz** [comment]
1. If you are using the core Image you could use something like `decodeWidth="50" decodeHeight="50" loadMode="async"` to help with the image loading or you can use one of the image cache libs available 
2. try to flatten your layout e.g single grid using cols, rows
3. use multiple templates if you are using directives with one of the frameworks e.g angular `*ngIf |  *ngFor`  , vue `v-if | v-for` or whatever the others use
